Frameworks/CoreGraphics/CGColorSpace.mm

```diff
@@ -38,135 +39,152 @@
 const CFStringRef kCGColorSpaceITUR_2020 = static_cast<CFStringRef>(@"kCGColorSpaceITUR_2020");
 const CFStringRef kCGColorSpaceROMMRGB = static_cast<CFStringRef>(@"kCGColorSpaceROMMRGB");
 
-static IWLazyClassLookup _LazyUIColor2("UIColor");
+struct __CGColorSpace : CoreFoundation::CppBase<__CGColorSpace> {
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace = nullptr, size_t lastIndex = 0)
+        : _colorSpaceModel(model), _baseColorSpace(baseSpace), _lastColorIndex(lastIndex) {
+    }
 
-@interface CGNSColorSpace : _CGLifetimeBridgingType
-@end
+    __CGColorSpace(CGColorSpaceModel model, CGColorSpaceRef baseSpace, size_t lastIndex, const unsigned char* colorTable)
+        : __CGColorSpace(model, baseSpace, lastIndex) {
+        size_t size = CGColorSpaceGetNumberOfComponents(baseSpace) * (lastIndex + 1);
+        _palette.reset(new char[size]);
+        memcpy(_palette.get(), colorTable, size);
+    }
 
-@implementation CGNSColorSpace
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wobjc-missing-super-calls"
-- (void)dealloc {
-    delete (__CGColorSpace*)self;
-}
-#pragma clang diagnostic pop
-@end
+    inline void SetColorSpaceModel(CGColorSpaceModel model) {
```

Comment:
leaving it as it could be useful.

---
In reply to: [106515394](https://github.com/Microsoft/WinObjC/pull/2253#discussion_r106515394) [](ancestors = 106515394)